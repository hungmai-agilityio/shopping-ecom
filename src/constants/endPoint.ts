export const END_POINT = {
  HOME: '/',
  ABOUT_US: '/about',
  CONTACT: '/contact',
  SIGN_IN: '/sign-in',
  SIGN_UP: '/sign-up',
  WISHLIST: '/wishlist',
  PRODUCTS: '/products',
  PRODUCT: '/product',
  CATEGORY: '/category',
  USERS: '/users',
  ACCOUNT: '/account',
  CART: '/cart',
  CHECKOUT: '/checkout'
};

export const QUERY = {
  CART: 'cart',
  USER: 'users',
  WISHLIST: 'wishlist',
  PRODUCTS: 'products',
}