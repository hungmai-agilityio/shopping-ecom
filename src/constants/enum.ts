export enum TYPE {
  PRIMARY = 'primary',
  SECOND = 'second',
  THIRD = 'third'
}

export enum INPUT_TYPE {
  FILE = 'file',
  TEXT = 'text',
  PASSWORD = 'password',
  EMAIL = 'email'
}

export enum SIZE {
  SMALL = 'small',
  MEDIUM = 'medium',
  LARGE = 'large'
}

export enum CELL {
  TD = 'td',
  TH = 'th'
}

export enum KEY {
  USER = 'user'
}

export enum TIME_STATUS {
  STALE_TIME = 1000 * 60 * 60, // milliseconds * seconds * minutes = 1 hour
  COOKIE_TIME = 60 * 60 * 24 * 2 // seconds * minutes * hour * day
}
