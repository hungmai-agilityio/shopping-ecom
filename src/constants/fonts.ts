import { Poppins, Roboto } from 'next/font/google';

export const popping = Poppins({
  subsets: ['latin'],
  weight: ['400', '500', '700']
});

export const roboto = Roboto({
  subsets: ['latin'],
  weight: ['400', '500', '700']
});
