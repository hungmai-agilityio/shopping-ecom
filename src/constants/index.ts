export * from '@/constants/enum';
export * from '@/constants/styles';
export * from '@/constants/fonts';
export * from '@/constants/endPoint';
export * from '@/constants/list';
