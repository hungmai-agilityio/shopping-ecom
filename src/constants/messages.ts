export const MESSAGE_VALID = {
  INPUT_REQUIRED: '{field} is required',
  EMAIL_ERROR: 'Invalid email format',
  EMAIL_EXIST: 'Email already exists',
  PASSWORD_ERROR: 'Password must be at least 6 characters',
  CONFIRM_ERROR: 'Passwords do not match',
  NEW_PASS_ERROR: 'Password cannot be the same as the old password',
  UPLOAD: 'Only file .png, .jpg, .jpeg, .svg allowed',
};

export const MESSAGE_API = {
  SIGN_UP_ERROR: 'An error occurred. Please try again',
  SIGN_UP_SUCCESS: 'You have successfully registered',
  SIGN_IN_ERROR: 'email or password is incorrect',
  ADD_PRODUCT_SUCCESS: 'Add product success',
  UPDATE_PROFILE_SUCCESS: 'Update profile success',
  UPDATE_PROFILE_ERROR: 'Update profile error',
  DELETE_ADDRESS_SUCCESS: 'Delete address success',
  DELETE_ADDRESS_ERROR: 'Delete address error',
  UPDATE_CART_ERROR: 'Update cart error',
  UPDATE_CART_SUCCESS: 'Update cart success',
  UPDATE_QUANTITY: 'Add quantity +1 to existing products'
};
