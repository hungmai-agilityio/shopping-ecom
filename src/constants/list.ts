import { END_POINT } from "@/constants/endPoint";

export const navItems = [
  { name: 'Home', url: END_POINT.HOME },
  { name: 'Contact', url: END_POINT.CONTACT },
  { name: 'About', url: END_POINT.ABOUT_US }
];

export const listAbout = [
  { name: 'My Account', url: END_POINT.PROFILE },
  { name: 'Login / Register', url: END_POINT.SIGN_IN },
  { name: 'Cart', url: END_POINT.CART },
  { name: 'Wishlist', url: END_POINT.WISHLIST }
]

export const listQuick = [
  { name: 'Privacy Policy', url: '' },
  { name: 'Terms Of Use', url: '' },
  { name: 'FAQ', url: '' },
  { name: 'Contact', url: END_POINT.CONTACT }
]

export const slideHero = [
  {
    id: 'hero-1',
    image: '/hero_iphone.png',
    content: 'Up to 10% off Voucher',
    title: 'IPhone 14 Series',
    icon: '/icon-iphone.svg'
  },
  {
    id: 'hero-2',
    image: '/hero-canon.png',
    content: 'Capture stunning photos every time.',
    title: 'Canon camera'
  },
  {
    id: 'hero-3',
    image: '/hero-shoes.png',
    content: 'Durable, stylish, all-day comfort',
    title: 'Adidas Sneakers'
  },
  {
    id: 'hero-4',
    image: '/hero-cooler.png',
    content: 'Max Cooling Power - Liquid Cooling for Your PC!',
    title: 'Advanced Liquid Cooling System'
  },
  {
    id: 'hero-5',
    image: '/hero-ps5.png',
    content: 'Level Up with PS5 - Buy Now',
    title: 'PS5',
    icon: '/logo-sony.svg'
  }
];

export const slideStaff = [
  {
    id: 'staff-1',
    image: '/lead.png',
    name: 'Tom Cruise',
    desc: 'Founder & Chairman',
  },
  {
    id: 'staff-1',
    image: '/manager.png',
    name: 'Emma Watson',
    desc: 'Managing Director',
  },
  {
    id: 'staff-1',
    image: '/designer.png',
    name: 'Will Smith',
    desc: 'Product Designer',
  }
]