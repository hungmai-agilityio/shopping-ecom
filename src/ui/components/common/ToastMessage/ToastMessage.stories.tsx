import { Meta, StoryObj } from '@storybook/react';
import ToastMessage from '.';
import { POSITION, STATUS } from '@/constants';

const meta: Meta<typeof ToastMessage> = {
  component: ToastMessage,
  tags: ['autodocs'],
  argTypes: {
    status: {
      description:
        'Set status for Toast message, it will show UI with corresponding styles',
      control: 'radio',
      options: Object.values(STATUS)
    },
    message: {
      description: 'Message to show'
    },
    duration: {
      description: 'Set the amount of time for Toast to be invisible'
    },
    position: {
      description: 'Set the position for Toast to display',
      control: 'radio',
      options: Object.values(POSITION)
    }
  }
};

export default meta;

type Story = StoryObj<typeof ToastMessage>;

export const success: Story = {
  args: {
    status: STATUS.SUCCESS,
    message: 'show message success',
    position: POSITION.TOP_RIGHT,
    duration: 100000
  },
  render: (args) => (
    <div className="w-[500px] h-[200px]">
      <ToastMessage {...args} />
    </div>
  )
};

export const error: Story = {
  args: {
    status: STATUS.ERROR,
    message: 'show message error',
    position: POSITION.TOP_RIGHT,
    duration: 100000
  },
  render: (args) => (
    <div className="w-[500px] h-[200px]">
      <ToastMessage {...args} />
    </div>
  )
};

export const warning: Story = {
  args: {
    status: STATUS.WARNING,
    message: 'show message warning',
    position: POSITION.TOP_RIGHT,
    duration: 100000
  },
  render: (args) => (
    <div className="w-[500px] h-[200px]">
      <ToastMessage {...args} />
    </div>
  )
};
