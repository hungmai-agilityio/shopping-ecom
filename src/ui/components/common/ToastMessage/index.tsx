'use client';

import { clsx } from 'clsx';
import { useEffect, useState } from 'react';

// Constants
import { POSITION, STATUS, toastStyles } from '@/constants';

interface ToastProps {
  status: STATUS.SUCCESS | STATUS.ERROR | STATUS.WARNING;
  message: string;
  duration?: number;
  position?:
    | POSITION.TOP_LEFT
    | POSITION.TOP_RIGHT
    | POSITION.BOT_LEFT
    | POSITION.BOT_RIGHT;
}

const ToastMessage = ({
  status,
  message,
  duration = 3000,
  position
}: ToastProps) => {
  const [visible, setVisible] = useState<boolean>(true);

  const positionClass = {
    [POSITION.TOP_RIGHT]: 'top-4 right-4',
    [POSITION.TOP_LEFT]: 'top-4 left-4',
    [POSITION.BOT_RIGHT]: 'bottom-4 right-4',
    [POSITION.BOT_LEFT]: 'bottom-4 left-4'
  };

  const setPosition = positionClass[position || POSITION.TOP_RIGHT];

  useEffect(() => {
    const timer = setTimeout(() => {
      setVisible(false);
    }, duration);

    return () => clearTimeout(timer);
  }, []);

  if (!visible) return null;

  return (
    <div
      className={clsx(
        'text-white font-semibold fixed py-2 px-4 rounded shadow-lg transition-opacity ease-in-out max-w-lg truncate',
        toastStyles[status],
        setPosition
      )}
    >
      {message}
    </div>
  );
};

export default ToastMessage;
