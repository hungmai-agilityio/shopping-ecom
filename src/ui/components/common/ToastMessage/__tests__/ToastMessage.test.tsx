import { render, screen } from '@testing-library/react';

// Components
import ToastMessage from '@/ui/components/common/ToastMessage';
import { STATUS } from '@/constants';

describe('ToastMessage component', () => {
  const defaultProps = {
    status: STATUS.SUCCESS,
    message: 'Messages success'
  };

  test('Should be render with default props', () => {
    render(<ToastMessage {...defaultProps} />);
    const message = screen.getByText('Messages success');
    expect(message).toBeInTheDocument();
    expect(message).toHaveClass('bg-green-500');
  });

  test('Should be render with custom status props', () => {
    render(
      <ToastMessage
        {...defaultProps}
        status={STATUS.ERROR}
        message="Messages error"
      />
    );
    const message = screen.getByText('Messages error');
    expect(message).toBeInTheDocument();
    expect(message).toHaveClass('bg-red-500');
  });

  test('Should be render match to snapshot', () => {
    const { container } = render(<ToastMessage {...defaultProps} />);

    expect(container).toMatchSnapshot();
  });
});
