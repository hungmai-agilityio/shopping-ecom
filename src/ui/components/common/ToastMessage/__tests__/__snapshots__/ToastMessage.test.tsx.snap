// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToastMessage component Should be render match to snapshot 1`] = `
<div>
  <div
    class="text-white font-semibold fixed py-2 px-4 rounded shadow-lg transition-opacity ease-in-out max-w-lg truncate bg-green-500 top-4 right-4"
  >
    Messages success
  </div>
</div>
`;
